lib/pure/parseopt.nim

```diff
@@ -32,9 +32,11 @@ type
     cmdShortOption            ## a short option ``-c`` detected
   OptParser* =
       object of RootObj ## this object implements the command line parser
-    cmd: string
-    pos: int
+    cmd*: string              #  cmd,pos exported so caller can catch "--" as..
+    pos*: int                 # ..empty key or subcmd cmdArg & handle specially
     inShortState: bool
+    shortNoArg: string
```

Comment:
this should be of type ``set[char]``. ``longNoArg`` should still be a seq to avoid the dependency I guess. Also ``@[]`` is easier to use.